src/Server/CertificateReloader.cpp

```diff
@@ -93,32 +109,57 @@ void CertificateReloader::tryLoad(const Poco::Util::AbstractConfiguration & conf
     }
     else
     {
-        bool cert_file_changed = cert_file.changeIfModified(std::move(new_cert_path), log);
-        bool key_file_changed = key_file.changeIfModified(std::move(new_key_path), log);
-        std::string pass_phrase = config.getString("openSSL.server.privateKeyPassphraseHandler.options.password", "");
+        auto it = data.end();
+        auto i = data_index.find(prefix);
+        if (i != data_index.end())
+            it = i->second;
+        else
+        {
+            data.push_back(MultiData(ctx));
+            --it;
+            data_index[prefix] = it;
+        }
+
+        bool cert_file_changed = it->cert_file.changeIfModified(std::move(new_cert_path), log);
+        bool key_file_changed = it->key_file.changeIfModified(std::move(new_key_path), log);
+        std::string pass_phrase = config.getString(prefix + "privateKeyPassphraseHandler.options.password", "");
```

Comment:
lets move this local variable to the more narrow scope inside `if (cert_file_changed || key_file_changed)`.